Version 1

Andrew-Fisenko opened the pull request.

Commit:
1.01
- Добавлен интерфейс с кнопками
- Работают кнопки с цифрами
- Цифры отображаются в первом поле для ввода - первое число
- Работает конка С - стирает введенное число

Commit:
1.02
- Добавлена ландшафтная ориентация

Commit:
1.03
- Полность работает функционал калькулятора на 4 заявленных операциях
- Работает кнопка С - сброс в любой момент процесса
- Повторное нажание на "Равно" повторяет предыдущую операцию
- Корретно работает при отсутствии какого из аргуметов при нажатии "Равно"
- Знак дейтвия виден и менятется при нажатии в любое время

Commit:
1.04
- код калькулятора разделен на два класса
CalculatorActivity - отвечает за все высичления и хранение информации
CalculatorActivity - за обработку действий кнопок и отображение результатов на экране

Commit:
1.05
- Добавлена возможность сохранять введенные данные при повороте после смены ориентации экрана
- Исправлена ошибка когда первы аргумент - цифра равен второму аргументу цифре

Commit:
1.06
- Стили использованы в виде тем для всего приложения
- Созданы светлая и темная темы
- Все размеры и строки перенесены в ресурсы

primus-anonymous commented on app/src/main/java/com/example/calculatorx/CalcuatorActivity.java:
```diff
@@ -0,0 +1,167 @@
+    public TextView txtOperation;
+
+    private CalculatorPresenter presenter;
+    private CalcuatorActivity calculatorView;
```
Зачем в CalcuatorActivity держать ссылку на CalcuatorActivity?

primus-anonymous commented on app/src/main/java/com/example/calculatorx/CalcuatorActivity.java:
```diff
@@ -0,0 +1,167 @@
+        this.calculatorView = calculatorView;
+    }
+
+    public CalcuatorActivity(TextView txtResult, TextView txtNumberOne, TextView txtNumberTwo, TextView txtOperation) {
```
Рекомендую пересмотреть вебинар про активити. Мы не создаем Activity сами, это делает Android из .class компонента, предоставляемого нами в intent'е. Не пустых конструкторов у Activity быть не может

primus-anonymous replied:
Даже чисто логически, все перечисленные view содержит Activity каким образом даже теоретически они могут поставляться зависимостью?

primus-anonymous commented on app/src/main/java/com/example/calculatorx/CalcuatorActivity.java:
```diff
@@ -0,0 +1,167 @@
+
+public class CalcuatorActivity extends AppCompatActivity {
+
+    public TextView txtResult;
```
Поля лучше сделать private, если нужен доступ к view то можно добавить getter'ов

primus-anonymous commented on app/src/main/java/com/example/calculatorx/CalcuatorActivity.java:
```diff
@@ -0,0 +1,167 @@
+        txtNumberTwo = findViewById(R.id.number_two);
+        txtOperation = findViewById(R.id.operation);
+
+        calculatorView = new CalcuatorActivity(txtResult, txtNumberOne, txtNumberTwo, txtOperation) {
```
Если здесь MVP, то CalcuatorActivity и есть view

primus-anonymous replied:
Т.е this, нет никакой необходимости создавать еще что-то

primus-anonymous commented on app/src/main/java/com/example/calculatorx/CalculatorPresenter.java:
```diff
@@ -0,0 +1,114 @@
+        if (numOne == 0 || numTwo == 0 || oper == null) {
+            return 0.0;
+        } else if (result != 0.0) {
+            if (oper == "/") {
```
Операнды, если их конкретное кол-во лучше вынести в enum

primus-anonymous commented on app/src/main/res/layout-land/activity_calculator.xml:
```diff
@@ -0,0 +1,272 @@
+
+
+
+
```
Многовато пустых строк, столько их тут не надо

Commit:
1.07
- Переведены в ресурсы значения размеров с макета ландшафтной ориентации экрана
- Доваблен Scroll на ландшафтную ориентацию
- Реализован выбор темы по нажатию
- При смене темы сохраняются введеные до этого данные
- Убрана часть лишнего кода
- При повороте экрана Тема приложения не сохраняется, а меняется на Default

Commit:
1.08
- Реализовано сохранение тем прилоения через storage
- Тема сохраняется втом числе и при смене ориентации
- Лишний код закомменчен, но не удален

Commit:
1.09
- Реализована смена темы приложения через нажатие на кнотки D и L
- Отключен список тем под калькулятором
- Убран Scroll на ландшафтной ориентации

Commit:
1.10
- Добавлена реализация кнопки "точка"
- Ограничена функциональность калькулятора до 3-х знаков после запятой в связи с появляющейся ошибкой ввода (например - 1.19999999999 вместо 1.2)
- Часть лишнего кода удалена, часть - закомментирована
- Это Финальная версия калькулятра - реализованы все изначально задуманные функции и возможности

Commit:
1.11
- Добавлен интент-фильтр для запуска калькулятора из другого приложения

Commit:
1.12
Исправил замечания из прошлого домашнего задания:
- Убрал ссылку на CalcuatorActivity
- убрал ненужные конструкторы activity из CalcuatorActivity
- сделал public поля private
- добавил CalcuatorActivity calculatorView = this;
- убрал пустые строки, выровнял код
- установил фоновое изображение вместо заливки фона цветом в обоих темах

primus-anonymous commented on app/src/main/java/com/example/calculatorx/CalcuatorActivity.java:
```diff
@@ -0,0 +1,256 @@
+        presenter = new CalculatorPresenter();
+
+            if (savedInstanceState != null) {
+            presenter.setNumOne(savedInstanceState.getDouble("numOne"));
```
Этот код явно нуждается в форматировании

primus-anonymous commented on app/src/main/java/com/example/calculatorx/CalcuatorActivity.java:
```diff
@@ -0,0 +1,256 @@
+        Button butDark = findViewById(R.id.but_dark);
+        Button butLight = findViewById(R.id.but_light);
+        for (Theme theme : Theme.values()) {
+            butDark.setOnClickListener(new View.OnClickListener() {
```
Выставление темы в отдельном активити необходимо для отработки activity for result, в проекте этого нет сейчас

primus-anonymous commented on app/src/main/java/com/example/calculatorx/CalcuatorActivity.java:
```diff
@@ -0,0 +1,256 @@
+            });
+        }
+
+//        for (Theme theme : Theme.values()) {
```
Не надо оставлять в коде кучу закомментированного кода. Первый же вопрос, это надо и случайно закомменчено или это не надо вообще?

primus-anonymous commented on app/src/main/java/com/example/calculatorx/CalculatorPresenter.java:
```diff
@@ -0,0 +1,162 @@
+            } else {
+                numTwo = numTwo + number / (double) divider;
+                divider *= base;
+                numTwo = toDecimalFormat(numTwo);
```
Это какое-то колдунство, здесь в принципе могут и теряться знаки после запятой
